src/auth/credential.ts

```diff
@@ -352,6 +352,21 @@ export function getApplicationDefault(httpAgent?: Agent): Credential {
     return credentialFromFile(process.env.GOOGLE_APPLICATION_CREDENTIALS, httpAgent);
   }
 
+  // Allow the Emulator Suite to pass an authorized_user credential as an environment
+  // variable. This is similar to the gcloud's application default credentials but
+  // not saved to the file system.
+  if (process.env.FIREBASE_EMULATOR_CREDENTIALS) {
+    try {
+      const creds = JSON.parse(process.env.FIREBASE_EMULATOR_CREDENTIALS);
```

Comment:
Do we also need to support specifying a path to a json file instead of the json string itself?

Reply:
No not yet, there is no such file it's something we construct at runtime inside th Functions emulator.